clvm-traits/src/match_byte.rs

```diff
@@ -0,0 +1,45 @@
+use clvmr::{
+    allocator::{NodePtr, SExp},
+    Allocator,
+};
+use num_bigint::{BigInt, Sign};
+
+use crate::{Error, FromClvm, Result, ToClvm};
+
+#[derive(Debug, Copy, Clone)]
+pub struct MatchByte<const BYTE: u8>;
+
+impl<const BYTE: u8> ToClvm for MatchByte<BYTE> {
+    fn to_clvm(&self, a: &mut Allocator) -> Result<NodePtr> {
+        a.new_number(BYTE.into()).map_err(Error::Allocator)
+    }
+}
+
+impl<const BYTE: u8> FromClvm for MatchByte<BYTE> {
+    fn from_clvm(a: &Allocator, node: NodePtr) -> Result<Self> {
+        if let SExp::Atom() = a.sexp(node) {
+            let value: u8 = to_unsigned(a.number(node)).try_into().map_err(
+                |error: <u8 as TryFrom<BigInt>>::Error| Error::Custom(error.to_string()),
+            )?;
```

Comment:
this construct makes me a bit uneasy, especially with the `to_unsigned` function that doesn't seem to do what its name suggests.

Unless I'm misunderstanding what you're doing, I think it would be much more reliable to ask for the `atom()` rather than the `number()` then ensure it's 1 byte long and then pull out that one byte.

However, you would need to take the special case of 0 into account, which is an empty atom.

Reply:
Improved the code here